src/xnnpack/simd/s32-avx512f.h

```diff
@@ -42,22 +42,59 @@ static XNN_INLINE xnn_simd_s32_t xnn_min_s32(xnn_simd_s32_t a,
   return _mm512_min_epi32(a, b);
 }
 
+static XNN_INLINE xnn_simd_s32_t xnn_clz_s32(xnn_simd_s32_t a) {
+  __m256i lowi = _mm512_extracti64x4_epi64(a, 0);
+  __m256i highi = _mm512_extracti64x4_epi64(a, 1);
+
+  __m512d low = _mm512_cvtepi32_pd(lowi);
+  __m512d high = _mm512_cvtepi32_pd(highi);
+
+  xnn_simd_s32_t low_a = _mm512_castpd_si512(low);
+  xnn_simd_s32_t high_a = _mm512_castpd_si512(high);
+
+  xnn_simd_s32_t shift_low = _mm512_srli_epi64(low_a, 52) & 0x7FF;
+  xnn_simd_s32_t shift_high = _mm512_srli_epi64(high_a, 52) & 0x7FF;
+
+  xnn_simd_s32_t mask =
+      _mm512_setr_epi32(1, 3, 5, 7, 9, 11, 13, 15, 0, 2, 4, 6, 8, 10, 12, 14);
+  xnn_simd_s32_t mlow = _mm512_permutexvar_epi32(mask, shift_low);
+  xnn_simd_s32_t mhigh = _mm512_permutexvar_epi32(mask, shift_high);
+
+  __m256i ai = _mm512_extracti64x4_epi64(mlow, 1);
+  __m256i bi = _mm512_extracti64x4_epi64(mhigh, 1);
+
+  xnn_simd_s32_t exponent =
+      _mm512_inserti64x4(_mm512_castsi256_si512(ai), bi, 1);
+
+  xnn_simd_s32_t result =
+      _mm512_sub_epi32(_mm512_set1_epi32(31),
+                       _mm512_sub_epi32(exponent, _mm512_set1_epi32(1023)));
+
+  xnn_simd_s32_t zero = _mm512_setzero_si512();
+
+  __mmask16 maskl = _mm512_cmpge_epi32_mask(a, zero);
+  result = _mm512_maskz_mov_epi32(maskl, result);
+  __mmask16 maskz = _mm512_cmpeq_epi32_mask(a, zero);
+  result = _mm512_mask_mov_epi32(result, maskz, _mm512_set1_epi32(32));
+  return result;
+}
+
 // Load/store operations.
 
 static XNN_INLINE xnn_simd_s32_t xnn_loadu_s32(const int32_t* ptr) {
-  return _mm512_loadu_epi32(ptr);
+  return _mm512_loadu_si512(ptr);
```

Comment:
Why are these changes needed?

Reply:
Without these, there is a mismatch in avx512 when we especially do bitwise operations.